docs/openid.md

```diff
@@ -89,7 +90,51 @@ oidc_config:
        localpart_template: "{{ user.name }}"
        display_name_template: "{{ user.name|capitalize }}"
 ```
+### [Keycloak][keycloak-idp]
 
+[Keycloak][keycloak-idp] is an opensource IdP maintained by Red Hat. 
+
+Follow the [Getting Started Guide](https://www.keycloak.org/getting-started) to install Keycloak and set up a realm.
+
+1. `Clients -> Create`
+
+2. Fill Fields
```

Comment:
This could use a little bit more, maybe referring to "Secure your first app" section of the getting started guide?

I think it could be something like:

"Create a new client, filling in the form with the values below:"

Additionally below instead of just "`Save`" It should say "Click save".

Reply:
I would argue against it because it would add confusion.

Step 1: Views this page
https://www.keycloak.org/docs/latest/server_admin/keycloak-images/clients.png

Step 2 and 3:  View a different page
https://www.keycloak.org/docs/latest/server_admin/keycloak-images/add-client-oidc.png

Reply:
For the credential tab to appear, `Access Type`  must be set to `confidential` and `saved`. Those save buttons are important to move to the next step.

Reply:
I'm not suggesting removing them, I'm suggesting expand the steps to give information about the actions instead of just the title of each button to press.

Reply:
Merging them will increase the confusion.

Reply:
https://github.com/matrix-org/synapse/pull/7659/commits/015444e40a490196481f68384ec1066a19f1195e#diff-de854b597751e2f32829a3fb90525167R99